modules/tinymce/src/themes/silver/test/ts/browser/editor/scrolling/EditorInScrollingContainerTest.ts

```diff
@@ -0,0 +1,861 @@
+import { ApproxStructure, Assertions, Waiter } from '@ephox/agar';
+import { beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun, Optional } from '@ephox/katamari';
+import { Class, Css, Html, Insert, InsertAll, Remove, SelectorFind, SugarBody, SugarElement, Traverse } from '@ephox/sugar';
+import { TinyDom, TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+interface TestUi {
+  readonly className: string;
+  readonly selector: string;
+}
+
+interface TestScenario {
+  readonly label: string;
+  readonly settings: {
+    readonly inline: boolean;
+    readonly shadow: 'inside' | 'outside' | 'none';
+  };
+}
+
+enum TriggerWaitDecision {
+  FailWithError,
+  KeepWaiting,
+  Succeed
+}
+
+describe('browser.tinymce.themes.silver.editor.scrolling.EditorInScrollingContainerTest', () => {
+  const sharedSettings = {
+    menubar: 'file',
+    toolbar: 'undo bold opendialog',
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  const fromClass = (className: string): TestUi => ({
+    className,
+    selector: `.${className}`
+  });
+
+  const ui = {
+    ancestors: {
+      scrollingWrapper: fromClass('scrolling-wrapper'),
+      outerScroller: fromClass('outer-scroller'),
+    },
+    editor: {
+      stickyHeader: fromClass('tox-editor-header'),
+
+      stickyHeaderVisible: fromClass('tox-editor-dock-fadein'),
+      stickyHeaderInvisible: fromClass('tox-editor-dock-fadeout'),
+      stickyHeaderTransitioning: fromClass('tox-editor-dock-transition'),
+      menuButton: fromClass('tox-mbtn'),
+      menu: {
+        className: 'tox-menu',
+        selector: '[role=menu]'
+      },
+      menubar: {
+        className: 'tox-menubar',
+        selector: '[role=menubar]'
+      }
+    }
+
+  };
+
+  const heights = {
+    banner: 300,
+    scroller: 500,
+    outerScroller: 200,
+    editor: 2000,
+    miniEditor: 400
+  };
+
+  const getAncestorUi = (editor: Editor, testUi: TestUi) => {
+    return SelectorFind.ancestor<HTMLElement>(TinyDom.container(editor), testUi.selector).getOrDie(
+      `Could not find selector in ancestors: ${testUi.selector}`
+    );
+  };
+
+  const getEditorUi = (editor: Editor, testUi: TestUi) => {
+    return SelectorFind.descendant<HTMLElement>(TinyDom.container(editor), testUi.selector).getOrDie(
+      `Could not find selector in descendants: ${testUi.selector}`
+    );
+  };
+
+  const resetScrolls = async (editor: Editor) => {
+    const scroller = getAncestorUi(editor, ui.ancestors.scrollingWrapper);
+    scroller.dom.scrollTo(0, 0);
+    window.scrollTo(0, 0);
+    editor.getWin().scrollTo(0, 0);
+
+    // Adding here to prevent flaky test, when scrolling to the top, test cases should'nt start with docked toolbar
```

Comment:
Is this needed for browsers/os that animate the scroll?

```suggestion
    // Added to prevent flaky test, when scrolling to the top, test cases shouldn't start with docked toolbar
```

Or I guess you could just leave the comment out since we have a lot of other places where we just add a waiter to avoid flakes without a comment. But if that comment gives you context that you need keep it.